### PR TITLE
Service introspection

### PR DESCRIPTION
**Pull request by ihasdapie**
This PR is part of a prototype implementation for https://github.com/ros-infrastructure/rep/pull/360 https://github.com/ros2/ros2/issues/1285. 

Changes:
- Add <service_event>_Event message generation for services and actions

### CODE REVIEW

**Comment by jacobperron** [comment]
rebased on rolling

File: rosidl_generator_py/resource/_msg_support.c.em [hunk 1]
```diff
@@ -119,7 +121,10 @@ if isinstance(member.type, AbstractNestedType) and isinstance(member.type.value_
 nested_header = '/'.join(type_[:-1] + ('detail', convert_camel_case_to_lower_case_underscore(type_[-1]),))
 nested_header += '__functions.h'
 }@
-@[    if nested_header in include_directives]@
+@[    if type_[-1].endswith(SERVICE_REQUEST_MESSAGE_SUFFIX) or type_[-1].endswith(SERVICE_RESPONSE_MESSAGE_SUFFIX)]
+@# Service request/response messages are included in the srv__struct
+@[continue]
```
Comment (ivanpauno) [inline rosidl_generator_py/resource/_msg_support.c.em:126]: we could use the same style as L128-129 to avoid confusions

**Review by ivanpauno — APPROVED** [review]
LGTM!

**Review by mjcarroll — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18017/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12543/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18652/)

**Comment by clalancette** [comment]
Another CI try with the latest style fixes:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18023/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12548/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18660/)

**Pull request merged**